src/stores/app.store.ts

```diff
@@ -0,0 +1,394 @@
+import bytes from 'bytes'
+import { defineStore } from 'pinia'
+
+import {
+  type AppConfig,
+  fetchAllApps,
+  fetchApp,
+  getThemeLogo,
+} from '@/services/gateway.service'
+import {
+  defaultUserLimit,
+  unlimitedUserLimit,
+  ChainMapping,
+  RegionMapping,
+  WalletMode,
+  type Chain,
+  type StorageRegion,
+  MAX_DATA_TRANSFER_BYTES,
+  api,
+} from '@/utils/constants'
+
+type UserLimitUnit = 'MB' | 'GB'
+type UserLimitTarget = 'storage' | 'bandwidth'
+
+type Logo = {
+  mode: 'dark' | 'light'
+  orientation: 'horizontal' | 'vertical'
+  url: string
+}
+
+type UserLimitState = {
+  isUnlimited: boolean
+  value?: number
+  unit?: UserLimitUnit
+}
+
+type UserLimitUnlimitedParam = {
+  type: UserLimitTarget
+  isUnlimited: boolean
+}
+
+type UserLimitParam = {
+  type: UserLimitTarget
+  value: number
+  unit: UserLimitUnit
+}
+
+type SocialAuthVerifier =
+  | 'google'
+  | 'twitter'
+  | 'twitch'
+  | 'reddit'
+  | 'github'
+  | 'discord'
+
+type SocialAuthField = 'clientId' | 'clientSecret' | 'redirectUri'
+
+type SocialAuthOption = {
+  name: 'Google' | 'Twitter' | 'Twitch' | 'Reddit' | 'Github' | 'Discord'
+  verifier: SocialAuthVerifier
+  hasClientSecret: boolean
+  hasRedirectUri: boolean
+  documentation: string
+}
+
+type SocialAuthUpdateParam = {
+  verifier: SocialAuthVerifier
+  field: SocialAuthField
+  value: string
+}
+
+type SocialAuthState = {
+  verifier: SocialAuthVerifier
+  clientId?: string
+  clientSecret?: string
+  redirectUri?: string
+}
+
+type Theme = 'light' | 'dark'
+
+interface AppState {
+  appName: string
+  appId: string
+  appAddress: string
+  logos: {
+    dark: {
+      horizontal: string
+      vertical: string
+    }
+    light: {
+      horizontal: string
+      vertical: string
+    }
+  }
+  access: {
+    selectedChain: Chain
+  }
+  auth: {
+    social: SocialAuthState[]
+    passwordless: {
+      javascriptOrigin: string
+      redirectUri: string
+    }
+    wallet: {
+      websiteDomain: string
+      selectedTheme: Theme
+      hasUIMode: boolean
+      hasUIModeInGateway: boolean
+    }
+    redirectUri: string
+  }
+  store: {
+    userLimits: {
+      storage: UserLimitState
+      bandwidth: UserLimitState
+    }
+    region: StorageRegion
+  }
+}
+
+function calculateUserLimits(userLimit: number): UserLimitState {
+  if (userLimit < MAX_DATA_TRANSFER_BYTES) {
+    const isUnder1GB = userLimit < bytes('1 GB')
+
+    const calculatedUserLimit = bytes(userLimit, {
+      unitSeparator: ' ',
+      unit: isUnder1GB ? 'MB' : 'GB',
+    })
+
+    const [value, unit] = calculatedUserLimit.split(' ') as [
+      number,
+      UserLimitUnit
+    ]
+    return { value, unit, isUnlimited: false }
+  } else {
+    return { isUnlimited: true }
+  }
+}
```

Comment:
Move this to utils

Reply:
Resolved at https://github.com/arcana-network/developer-dashboard/pull/119/commits/e6ab91a55b03a69772dd31b278c9f277636a5a1d